8/debian-10/rootfs/opt/bitnami/scripts/grafana/entrypoint.sh

```diff
@@ -14,13 +14,37 @@ set -o pipefail
 . /opt/bitnami/scripts/libbitnami.sh
 . /opt/bitnami/scripts/liblog.sh
 
+function is_exec() {
+    # This checks if the first provided argument is executable or if only args was used
+    exec_in_path=$(builtin type -P "$1" 2> /dev/null)
+    if [[ -f "$1" && -x $(realpath "$1" 2> /dev/null) || -x $(realpath "$exec_in_path" 2> /dev/null) ]]; then
+        true;
+    else
+        false;
+    fi;
+}
+
+
 print_welcome_page
 
 if [[ "$1" = "/opt/bitnami/scripts/grafana/run.sh" ]]; then
-    /opt/bitnami/scripts/grafana/setup.sh
+    # This catches the error-code from libgrafana.sh for the immediate exit when the grafana-operator is used. And ensure that the exit code is kept silently.
+    /opt/bitnami/scripts/grafana/setup.sh && GRAFANA_OPERATOR_IMMEDIATE_EXIT=$? || GRAFANA_OPERATOR_IMMEDIATE_EXIT=$?
+    if [ $GRAFANA_OPERATOR_IMMEDIATE_EXIT -ne 0 ] && [ $GRAFANA_OPERATOR_IMMEDIATE_EXIT -ne 255 ]; then
+        exit $GRAFANA_OPERATOR_IMMEDIATE_EXIT
+    fi
```

Comment:
This is a bit hard to read, I would suggest something like this instead: 

```suggestion
    /opt/bitnami/scripts/grafana/setup.sh || GRAFANA_OPERATOR_IMMEDIATE_EXIT=$?
    if [[ "${GRAFANA_OPERATOR_IMMEDIATE_EXIT:-0}" -ne 0 ]]; then
        exit $GRAFANA_OPERATOR_IMMEDIATE_EXIT
    fi
```

Reply:
I don‘t know if that is the same behavior as before.
Does this ensure that the exit code is `0` when `255` was raised?

Reply:
With the proposed improvement, it would exit whenever `GRAFANA_OPERATOR_IMMEDIATE_EXIT` has a value different than `0`. And if `setup.sh` does not fail, `GRAFANA_OPERATOR_IMMEDIATE_EXIT` will default to `0` so it will not exit.

Reply:
Oh, I see now. So yes, you are right, this would need to be changed like this:

```
if [[ "${GRAFANA_OPERATOR_IMMEDIATE_EXIT:-0}" -eq 255 ]]; then
    exit 0
elif [[ "${GRAFANA_OPERATOR_IMMEDIATE_EXIT:-0}" -ne 0 ]]; then
    exit "$GRAFANA_OPERATOR_IMMEDIATE_EXIT"
fi
```

Reply:
Jep I am also fine removing the additional check after the init scripts in favor of this solution :)

Reply:
@marcosbc I applied all you changes now.
Thanks for the help :)